ui/components/compose.jsx

```diff
@@ -115,25 +119,37 @@ module.exports = function(core, config, store) {
 			}
 		},
 
+		onStateChange: function(changes, next) {
+			if (changes.user || (changes.nav && changes.nav.currentText)) {
+				this.onChange(true);
+			}
+
+			next();
+		},
+
 		componentDidMount: function() {
-			this.setPlaceHolder();
+			this.focusInput();
+
+			core.on("statechange", this.onStateChange, 100);
```

Comment:
On global statechange the component doesn't update as the state of the component remains the same. The component only updates when it's state changes.

If I don't update the state on input, then if "setState" is called somewhere else, the text will get overwritten. I cannot determine if the newState.userInput should replace the entered text or not just by looking at them, also I cannot update the component state from within componentWillUpdate.

Not updating state onInput means the state and the component remain in inconsistent state, which can cause further issues.

I need to listen to onPaste as I need to strip the formatting of the pasted html.

I think it's always better to depend on a single state (either the component state or the global state) to update the component. The component state updates in various situations anyways, so I chose to make the component only update when its internal state changes. Makes the whole thing much simpler and easily to understand.